cmd/sovereignnode/incomingHeader/incomingHeaderProcessor_test.go

```diff
@@ -752,28 +752,31 @@ func TestIncomingHeaderHandler_AddHeader(t *testing.T) {
 		},
 	}
 
-	tcCalled := -1
-	args.TopicsChecker = &mock.TopicsCheckerMock{
+	checkValidityCt := -1
+	args.TopicsChecker = &sovTests.TopicsCheckerMock{
 		CheckValidityCalled: func(topics [][]byte) error {
-			tcCalled++
+			checkValidityCt++
 
-			if tcCalled == 0 {
+			switch checkValidityCt {
+			case 0:
 				require.Equal(t, topic1, topics)
-			} else {
+			case 1:
 				require.Equal(t, topic2, topics)
+			default:
+				require.Fail(t, "check validity called more than 2 times")
 			}
-
 			return nil
 		},
 	}
 
 	deserializeEventDataCt := -1
 	deserializeTokenDataCt := -1
-	args.DataCodec = &mock.DataCodecMock{
+	args.DataCodec = &sovTests.DataCodecMock{
 		DeserializeEventDataCalled: func(data []byte) (*sovereign.EventData, error) {
 			deserializeEventDataCt++
 
-			if deserializeEventDataCt == 0 {
+			switch deserializeEventDataCt {
+			case 0:
```

Comment:
I don't think these tests are correct.
Judging by the code, your switch never reaches 0 case everywhere else in the code